### PR TITLE
[OpInfo] Add opinfo for `transpose` and its aliases

### PR DESCRIPTION
**Pull request by kshitij12345**
Reference: #54261

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -4672,21 +4698,6 @@ def method_tests():
         ('float_power', torch.rand(S, S, S) + 1e-3, (uniform_scalar(0.1),), 'scalar_broadcast_rhs'),
         ('float_power', uniform_scalar(1e-3, requires_grad=True), (torch.rand(S, S, S) + 0.1,), 'scalar_broadcast_lhs'),
         ('float_power', torch.rand(S, S, S) + 1e-3, (3.14,), 'constant'),
-        ('transpose', (1, 2, 3), (1, 2), 'dim', (False,), [0, 1]),
```
Comment (kshitij12345) [inline torch/testing/_internal/common_methods_invocations.py:4675]: [0, 1] at the end corresponds to the following line
https://github.com/pytorch/pytorch/blob/f5a7b2e641a124751dfa3fdf1926eb4135d3aad5/test/test_autograd.py#L5390-L5393

For entry `('transpose', (1, 2, 3), (1, 2), 'dim', (False,), [0, 1])` creates tests with args being  `[(-1, -2), (-1, 2), (1, -2), (1, 2)]`

We directly add these to cases 
https://github.com/pytorch/pytorch/blob/d4875aa5dd6499b31e2a0d01466d0a2382004d6b/torch/testing/_internal/common_methods_invocations.py#L711-L718

Had added it with similar to machinery to test_autograd test generator but adding directly seemed more readable.
https://github.com/pytorch/pytorch/blob/7b089c4cd21b046fdae85d0cec62961327bd8cf2/torch/testing/_internal/common_methods_invocations.py#L711-L722

File: test/test_torch.py [hunk 2]
```diff
@@ -7144,7 +7144,6 @@ def tmp(dtype, device):
         lambda t, d: [_number(0.5, 3, t), _number(0.4, 2, t), _medium_2d(t, d), _medium_1d(t, d)], 1e-2, 1e-1, 1e-4,
         torch.testing.get_all_fp_dtypes(include_bfloat16=AMPERE_OR_ROCM) + _complex_types_skip_rocm, _cpu_types, True,
         [_wrap_warn_once("This overload of addmv_? is deprecated")]),
-    ('atan2', '', _medium_2d, lambda t, d: [_medium_2d(t, d)], 1e-2, 1e-5, 1e-5, _types, _types_no_half),
```
Comment (kshitij12345) [inline test/test_torch.py:7147]: Opinfo for atan2 was added in https://github.com/pytorch/pytorch/pull/55132

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (mruberry) [reply torch/testing/_internal/common_methods_invocations.py:4675]: I agree with you it's simpler to add these directly for now. Thanks for paying close attention here. This generation is well hidden by method_tests(). If we do end up with similar generation at least it will be explicit in the sample input function.

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -704,6 +704,26 @@ def sample_inputs_trace(self, device, dtype, requires_grad, **kwargs):
...
+def sample_inputs_transpose_swapdims(self, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
+
+    cases = [((1, 2, 3), (-1, -2)),
+             ((1, 2, 3), (-1, 2)),
+             ((1, 2, 3), (1, -2)),
+             ((1, 2, 3), (1, 2)),
+             ((), (0, 0)),
+             ((1, ), (0, 0)),
+             ((L, L), (0, 1)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:717]: L, L -> M, M?

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -704,6 +704,26 @@ def sample_inputs_trace(self, device, dtype, requires_grad, **kwargs):
                                      low=None, high=None,
                                      requires_grad=requires_grad))),)
 
+
+def sample_inputs_transpose_swapdims(self, device, dtype, requires_grad, **kwargs):
+    make_arg = partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
+
+    cases = [((1, 2, 3), (-1, -2)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:711]: nit: prefer tuples to lists when the datastructure doesn't need to be mutated

**Review by mruberry — APPROVED** [review]
Nice work on the sample inputs, @kshitij12345, and it's great to see more aliases ported (I hope we can get rid of test_op_aliases.py soon!). 

I made a couple super small suggestions but this just needs to rebased. Ping me when it's ready to merge.

File: torch/testing/_internal/common_methods_invocations.py (hunk 3 above)
Comment (kshitij12345) [reply torch/testing/_internal/common_methods_invocations.py:717]: Updating to (M, M)
**Note**: Single run timings (very high variability)

Timings with (S, S)
```
=================================================== 62 passed, 35 skipped, 14505 deselected, 5 warnings in 23.48s ====================================================
```

Timings with (M, M)
```
==================================================== 62 passed, 35 skipped, 14505 deselected, 5 warnings in 23.21s =====================================================
```

Timings with (L, L)
```
==================================================== 62 passed, 35 skipped, 14505 deselected, 5 warnings in 29.23s =====================================================
```

**Comment by kshitij12345** [comment]
@mruberry Gentle Ping :)

Failure looks irrelevant : [link](https://app.circleci.com/pipelines/github/pytorch/pytorch/304297/workflows/6a7e50aa-3746-4911-8ac7-e7785ae17381/jobs/12546822)
